src/blas/backends/syclblas/CMakeLists.txt

```diff
@@ -0,0 +1,197 @@
+#==========================================================================
+#  Copyright (C) Codeplay Software Limited
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  For your convenience, a copy of the License has been included in this
+#  repository.
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#=========================================================================
+
+set(LIB_NAME onemkl_blas_syclblas)
+set(LIB_OBJ ${LIB_NAME}_obj)
+
+option(SYCLBLAS_TUNING_TARGET "Set a TUNING_TARGET for SYCL-BLAS" "")
+
+# Parse compiler flags and return a list of SYCL targets
+# The list is empty if no targets are set
+function(get_sycl_targets FLAGS)
+  string(REGEX MATCH "-fsycl-targets=[^ ]*" SYCL_TARGETS_FLAG "${FLAGS}")
+  string(REPLACE "-fsycl-targets=" "" SYCL_TARGETS "${SYCL_TARGETS_FLAG}")
+  string(REPLACE "," ";" SYCL_TARGETS "${SYCL_TARGETS}")
+  set(SYCL_TARGETS ${SYCL_TARGETS} PARENT_SCOPE)
+endfunction(get_sycl_targets)
+
+# SYCL-BLAS supports tuning for some device types, but can only be compiled
+# for one at a time currently. Work out which device to tune for based on the
+# DPC++ target triple specified via -fsycl-targets
+if(TARGET ONEMKL::SYCL::SYCL)
+  get_target_property(ONEMKL_COMPILE_OPTIONS ONEMKL::SYCL::SYCL COMPILE_OPTIONS)
```

Comment:
With cmake 3.16 this gives
```
INTERFACE_LIBRARY targets may only have whitelisted properties. The property "COMPILE_OPTIONS" is not allowed.
```
It works fine with cmake 3.20 which is what I usually use, but I think 3.16 is standard for our testing/CI. Do you know if there's an easy way to fix this line for earlier cmake versions?

Reply:
Thanks for testing this. I pushed a fix for this and confirmed CMake 3.16 now works.